models/surveys.go

```diff
@@ -29,53 +126,177 @@ func AllSurveys() ([]*SurveySummary, error) {
-		return nil, err
+	if len(surveySummaries) == 0 {
+		http.Error(w, "No surveys found", http.StatusNoContent)
+		return
 	}
-
-	return surveySummaries, nil
+	data, _ := json.Marshal(surveySummaries)
+	w.Write(data)
 }
 
 // GetSurvey returns the details of the survey identified by the string surveyID.
-func GetSurvey(surveyID string) (*Survey, error) {
+func (api *API) GetSurvey(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["surveyId"]
 	survey := new(Survey)
-	err := db.QueryRow("SELECT id, shortname, longname, surveyref from survey.survey WHERE id = $1", surveyID).Scan(&survey.ID, &survey.ShortName, &survey.LongName, &survey.Reference)
+	surveyRow := api.GetSurveyStmt.QueryRow(id)
+	err := surveyRow.Scan(&survey.ID, &survey.ShortName, &survey.LongName, &survey.Reference)
 	if err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			http.Error(w, "Survey not found", http.StatusNotFound)
+		} else {
+			http.Error(w, "get survey query failed", http.StatusInternalServerError)
+		}
 	}
 
-	return survey, nil
+	data, _ := json.Marshal(survey)
+	w.Write(data)
 }
 
 // GetSurveyByShortName returns the details of the survey identified by the string shortName.
-func GetSurveyByShortName(shortName string) (*Survey, error) {
+func (api *API) GetSurveyByShortName(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["shortName"]
+	surveyRow := api.GetSurveyByShortNameStmt.QueryRow(id)
 	survey := new(Survey)
-	err := db.QueryRow("SELECT id, shortname, longname, surveyref from survey.survey WHERE LOWER(shortName) = LOWER($1)", shortName).Scan(&survey.ID, &survey.ShortName, &survey.LongName, &survey.Reference)
+	err := surveyRow.Scan(&survey.ID, &survey.ShortName, &survey.LongName, &survey.Reference)
 	if err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			http.Error(w, "Survey not found", http.StatusNotFound)
+		} else {
+			http.Error(w, "get survey by shortname query failed", http.StatusInternalServerError)
+		}
 	}
 
-	return survey, nil
+	data, _ := json.Marshal(survey)
+	w.Write(data)
+
 }
 
-// GetSurveyByReference returns the details of the survey identified by the string reference.
-func GetSurveyByReference(reference string) (*Survey, error) {
+// GetSurveyByReference returns the details of the survey identified by the string ref.
+func (api *API) GetSurveyByReference(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["ref"]
+	surveyRow := api.GetSurveyByReferenceStmt.QueryRow(id)
 	survey := new(Survey)
-	err := db.QueryRow("SELECT id, shortname, longname, surveyref from survey.survey WHERE LOWER(surveyref) = LOWER($1)", reference).Scan(&survey.ID, &survey.ShortName, &survey.LongName, &survey.Reference)
+	err := surveyRow.Scan(&survey.ID, &survey.ShortName, &survey.LongName, &survey.Reference)
 	if err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			http.Error(w, "Survey not found", http.StatusNotFound)
```

Comment:
The previous version of the survey service had a "RESTError" model to represent a 404 not found - i think other services calling expect this data structure so we need to use it for the time being